packages/server/lib/browsers/electron.ts

```diff
@@ -384,6 +384,15 @@ export = {
     // adds a header to the request to mark it as a request for the AUT frame
     // itself, so the proxy can utilize that for injection purposes
     win.webContents.session.webRequest.onBeforeSendHeaders((details, cb) => {
+      const requestModifications = {
+        requestHeaders: {
+          ...details.requestHeaders,
+          ...(details.resourceType === 'xhr') ? {
```

Comment:
Is the `resourceType` for fetch also "xhr"? I notice we're checking for both "xhr" and "fetch" in CDP, but only "xhr" in electron and the extension.

Reply:
It has to do with electron and the extension only supporting `xhr`, meaning that `xhr` and `fetch` requests just show up as `xhr`.  But CDP can distinguish the difference. I'm going to add some docs here to explain why this is the way it is so its a bit clearer

Reply:
addressed in #24060